autowarefoundation/autoware.universe
# feat(default_ad_api): add interface version

Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

I started implementing the API prototype. I believe this helps discussion of API.

- Add interface vesion API.
- Add rclcpp service class wrapper for logging.
- Add message types.
- Add sample HTTP server.

## Related links

https://github.com/autowarefoundation/autoware-documentation/pull/69

## Tests performed

1. Launch the service server.
    `ros2 component standalone default_ad_api default_ad_api::InterfaceVersionNode`
2. Call the service by another terminal.
    `ros2 service call /api/interface/version autoware_ad_api_msgs/srv/InterfaceVersion "{}"`
3. Check the service response.
    `version = 0.5.0` and `response.summary.code = SUCCESS`
4. Check service server log.
    `service call: /api/interface/version ...` and `service exit: /api/interface/version ...`
5. Check HTTP server.
    Open a browser and access to `http://localhost:8888/interface/version`

## Notes for reviewers

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
